connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -167,4 +169,26 @@ message Expression {
     // (Optional) Alias metadata expressed as a JSON map.
     optional string metadata = 3;
   }
+
+  message LambdaFunction {
```

Comment:
I am on the fence about adding this message. We could go the CaseWhen route here as well. The only argument against that are the arguments them self :).

Reply:
> How would the lambda function look like for using the `CaseWhen` route?

this one https://github.com/apache/spark/pull/38956, we build the expressions in Connect Planner other than let `FunctionRegistry` do this

Reply:
actually, we can remove `ExpressionString`, `UnresolvedStar`, `UnresolvedRegex`, `Cast` in the same way.

But now I'm not sure whether it is the correct way.

Reply:
I think this is tricky because it's hard to provide clear guidance on what the right approach is. Generally, the downside of the unresolved function approach is that you're using a magic value that someone has to understand. This knowledge is now embedded in the client and cannot be inferred when looking at the protos.

Reply:
will add LambdaFunction back in the protos

Reply:
> I think this is tricky because it's hard to provide clear guidance on what the right approach is. Generally, the downside of the unresolved function approach is that you're using a magic value that someone has to understand. This knowledge is now embedded in the client and cannot be inferred when looking at the protos.

Agreed, that is what I feel. We should avoid abusing unresolved function